File: csharp/Gherkin/Ast/ScenarioDefinition.cs
using System;
using System.Linq;

namespace Gherkin.Ast
{
	public abstract class ScenarioDefinition : IHasLocation, IHasDescription, IHasSteps, IHasTags
	{
		public Tag[] Tags { get; private set; }
		public Location Location { get; private set; }
		public string Keyword { get; private set; }
		public string Title { get; private set; }
		public string Description { get; private set; }
		public Step[] Steps { get; private set; }

		protected ScenarioDefinition(Tag[] tags, Location location, string keyword, string title, string description, Step[] steps)
		{
			Tags = tags;
			Location = location;
			Keyword = keyword;
			Title = title;
			Description = description;
			Steps = steps;
		}
	}
}

File: src/main/java/gherkin/ast/ScenarioDefinition.java
package gherkin.ast;

import java.util.List;

public abstract class ScenarioDefinition implements HasDescription, HasSteps {
    private final List<Tag> tags;
    private final Location location;
    private final String keyword;
    private final String title;
    private final String description;
    private final List<Step> steps;

    public ScenarioDefinition(List<Tag> tags, Location location, String keyword, String title, String description, List<Step> steps) {

        this.tags = tags;
        this.location = location;
        this.keyword = keyword;
        this.title = title;
        this.description = description;
        this.steps = steps;
    }

    @Override
    public String getTitle() {
        return title;
    }

    @Override
    public List<Step> getSteps() {
        return steps;
    }

    public List<Tag> getTags() {
        return tags;
    }

    public Location getLocation() {
        return location;
    }

    @Override
    public String getKeyword() {
        return keyword;
    }

    @Override
    public String getDescription() {
        return description;
    }
}
